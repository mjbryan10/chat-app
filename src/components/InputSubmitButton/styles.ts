import styled, { ThemeProps, Theme } from 'styled-components';

export const Input = styled.input`
   background: rgb(249, 212, 35);
   background: ${({ theme: { background } }: ThemeProps<Theme>) =>
      background.buttonPrimary ?? 'inherit'};
   border: ${({ theme }: ThemeProps<Theme>) => theme.border};
   padding: 1em;
   margin: 0 0.3rem;
   &:hover {
      cursor: pointer;
   }
`;
