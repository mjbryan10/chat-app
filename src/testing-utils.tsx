import React from 'react';
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { dark } from './shared/theme';

const AllTheProviders = ({ children }: any) => {
  return (
    <ThemeProvider theme={dark}>
        {children}
    </ThemeProvider>
  )
  
}

const customRender = (ui?: any, options?: any) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }