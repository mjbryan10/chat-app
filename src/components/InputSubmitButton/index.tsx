import React, { SyntheticEvent } from 'react';
import * as S from './styles';

interface Props {
   value: string;
   disabled?: boolean;
   handleClick?: () => void;
}

const InputSubmitButton: React.FC<Props> = ({ value, disabled = false, handleClick }) => {
   const onClick = (event: SyntheticEvent<HTMLInputElement>) => {
      if (handleClick) {
         event.preventDefault();
         handleClick();
      }
   };
   return (
      <S.Input
         type="submit"
         value={value}
         data-testid="input-submit-button"
         disabled={disabled}
         onClick={onClick}
      />
   );
};

export default InputSubmitButton;
