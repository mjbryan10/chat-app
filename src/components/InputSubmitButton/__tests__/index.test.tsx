import React from 'react';
import InputSubmitButton from '..';
import { render, fireEvent, cleanup } from 'testing-utils';

describe('InputSubmitButton', () => {
   const testString = 'Given text';

   afterEach(() => {
      cleanup();
   });

   test('should have value passed from props', () => {
      const { getByTestId } = render(<InputSubmitButton value={testString} />);
      const input = getByTestId('input-submit-button') as HTMLInputElement;

      expect(input.value).toBe(testString);
   });

   test('should be disabled if disabled prop is passed', () => {
      const { getByTestId } = render(<InputSubmitButton value={testString} disabled />);

      const input = getByTestId('input-submit-button') as HTMLInputElement;

      expect(input).toBeDisabled();
   });

   test('should submit a form if nested and no handler passed', () => {
      const __onSubmit = jest.fn(e => e.preventDefault());
      const { getByTestId } = render(
         <form onSubmit={__onSubmit}>
            <InputSubmitButton value={testString} />
         </form>
      );

      fireEvent.click(getByTestId('input-submit-button'));

      expect(__onSubmit).toBeCalledTimes(1);
   });

   test('should call click handler if passed one in props', () => {
      const __handleClick = jest.fn();
      const { getByTestId } = render(
            <InputSubmitButton value={testString} handleClick={__handleClick} />
      );

      fireEvent.click(getByTestId('input-submit-button'));

      expect(__handleClick).toBeCalledTimes(1);
   });

   test('should render correctly', () => {
      const { asFragment } = render(<InputSubmitButton value={testString} />);

      expect(asFragment()).toMatchSnapshot();
   });
});
