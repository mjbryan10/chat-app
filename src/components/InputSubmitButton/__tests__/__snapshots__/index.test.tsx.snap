// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSubmitButton should render correctly 1`] = `
<DocumentFragment>
  <input
    class="sc-AxjAm ernPVN"
    data-testid="input-submit-button"
    type="submit"
    value="Given text"
  />
</DocumentFragment>
`;
